app/routes.py

```diff
@@ -1,2 +1,170 @@
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f'Task {task.task_id} "{task.title}" successfully deleted'}, 200)
+    
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+    task.completed_at = datetime.now()
+
+    path = 'https://slack.com/api/chat.postMessage'
+
+    query_dictionary= {
+        "token" : os.environ.get("SLACK_BOT_TOKEN"),
+        "channel" : "task-notifications",
+        "text" : f"Someone just completed the task {task.title}"
+    }
+    requests.post(path, data = query_dictionary)
+    
+    db.session.commit()
+    return make_response({"task": task.return_task_json()}, 200)
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def task_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404) 
+    task.completed_at = None
+    return make_response({"task": task.return_task_json()}, 200)
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response({"details": "Invalid data"}, 400) 
+
+    goal = Goal(title=request_body["title"])
+    
+    db.session.add(goal)
+    db.session.commit()
+
+    return make_response(goal.goal_json(), 201)
+
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_tasks():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        goals_response.append({
+            "id" : goal.goal_id,
+            "title" : goal.title
+        })
+    return jsonify(goals_response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def get_single_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    form_data = request.get_json()
+    if goal is None:
+        return make_response("", 404)
+    if request.method == "GET":
+        return make_response(goal.goal_json())
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        goal.title = form_data["title"]
+        db.session.commit()
+        return make_response(goal.goal_json())
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'}, 200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def goals_and_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    
+    request_body = request.get_json()
+    for task_id in request_body["task_ids"]:
+        task = Task.query.get(task_id)
+        task.goal_id = goal.goal_id
+
+    db.session.commit()
+    return make_response({"id": goal.goal_id, "task_ids": request_body["task_ids"]}), 200
```

Comment:
This change would have made the above-mentioned bug more obvious. This would also allow the response to include ALL tasks associated with the goal, not just the new ones.
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1

```suggestion
    full_task_ids = []
    for task in goal.tasks:
        full_task_ids.append(task.task_id)
        
    return make_response({"id": goal.goal_id, "task_ids": full_task_ids}, 200)
```